turbo/engineapi/fork_validator.go

```diff
@@ -256,10 +256,30 @@ func (fv *ForkValidator) ValidatePayload(tx kv.RwTx, header *types.Header, body
 	if unwindPoint == fv.currentHeight {
 		unwindPoint = 0
 	}
-	// if it is not canonical we validate it in memory and discard it aferwards.
-	batch := memdb.NewMemoryBatch(tx)
-	defer batch.Close()
-	return fv.validateAndStorePayload(batch, header, body, unwindPoint, headersChain, bodiesChain)
+	status, latestValidHash, validationError, criticalError = fv.validateAndStorePayload(tx, header, body, unwindPoint, headersChain, bodiesChain)
+	if criticalError != nil {
+		return
```

Comment:
hiding errors is not good, at least let's add a warning log here

Reply:
We are not hiding errors we are still returning it